File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque
input = sys.stdin.readline
N = int(input())
q = deque()

for i in range(N):
    l = list(input().rstrip().split())
    if l[0] == 'push':
        q.append(l[1])
    elif l[0] == 'front':
        if q:
            print(q[0])
        else:
            print(-1)
    elif l[0] == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)
    elif l[0] == 'size':
        print(len(q))
    elif l[0] == 'pop':
        if q:
            now = q.popleft()
            print(now)
        else:
            print(-1)
    else:
        if q:
            print(0)
        else:
            print(1)

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>


using namespace std;


int main() {

	int N;
	cin >> N;
	queue<int> q;

	string order;
	int num;
	for (int i = 0;i < N;i++) {
		cin >> order;
		if (order == "push") {
			cin >> num;
			q.push(num);
		}
		else if (order == "pop") {
			if (q.empty()) {
				cout << -1<<'\n';
			}
			else {
				cout << q.front() << '\n';
				q.pop();
			}
		}
		else if (order == "size") {
			cout << q.size() << '\n';
		}
		else if (order == "empty") {
			if (q.empty()) {
				cout << 1 << '\n';
			}
			else {
				cout << 0 << '\n';
			}
		}
		else if (order == "front") {
			if (q.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << q.front() << '\n';
			}
		}
		else {
			if (q.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << q.back() << '\n';
			}

		}

	}

	return 0;
}